src/main/java/cz/it4i/parallel/HeappeParadigm.java

```diff
@@ -0,0 +1,116 @@
+package cz.it4i.parallel;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
+
+import org.scijava.parallel.ParallelizationParadigm;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import cz.it4i.fiji.haas_java_client.HaaSClient;
+import cz.it4i.fiji.haas_java_client.JobSettingsBuilder;
+import cz.it4i.fiji.haas_java_client.JobState;
+import cz.it4i.fiji.haas_java_client.SettingsProvider;
+import cz.it4i.fiji.haas_java_client.TunnelToNode;
+
+@Plugin(type = ParallelizationParadigm.class)
+public class HeappeParadigm extends SimpleOstravaParadigm {
+
+	private static final int TIMEOUT_BETWEEN_JOB_STATE_QUERY = 1000;
+
+	@Parameter
+	private int port;
+
+	public static final Logger log = LoggerFactory.getLogger(cz.it4i.parallel.HeappeParadigm.class);
+
+	private int numberOfHosts;
+
+	private HaaSClient haasClient;
+	private final Collection<TunnelToNode> tunnels = Collections.synchronizedList(new LinkedList<>());
+
+	// -- HeappeParadigm methods --
+
+	public void setPort(int port) {
+		this.port = port;
+	}
+
+	public void setNumberOfNodes(int number) {
+		this.numberOfHosts = number;
+	}
+
+	// -- SimpleOstravaParadigm methods --
+
+	// -- Closeable methods --
```

Comment:
Closeable methods shall be defined further down in the file, see line 106.

Reply:
OK